app/src/main/java/com/cornellappdev/transit/ui/screens/DetailsScreen.kt

```diff
@@ -0,0 +1,161 @@
+import androidx.constraintlayout.compose.ConstraintLayout
+import androidx.navigation.NavHostController
+import com.cornellappdev.transit.R
+import com.cornellappdev.transit.models.RouteOptionType
+import com.cornellappdev.transit.networking.ApiResponse
+import com.cornellappdev.transit.ui.components.TransitPolyline
+import com.cornellappdev.transit.ui.theme.DividerGrey
+import com.cornellappdev.transit.ui.theme.IconGrey
+import com.cornellappdev.transit.ui.theme.MetadataGrey
+import com.cornellappdev.transit.ui.theme.PrimaryText
+import com.cornellappdev.transit.ui.theme.sfProDisplayFamily
+import com.cornellappdev.transit.ui.viewmodels.HomeViewModel
+import com.google.accompanist.permissions.ExperimentalPermissionsApi
+import com.google.accompanist.permissions.isGranted
+import com.google.accompanist.permissions.rememberPermissionState
+import com.google.android.gms.maps.model.CameraPosition
+import com.google.maps.android.compose.GoogleMap
+import com.google.maps.android.compose.MapProperties
+import com.google.maps.android.compose.MapUiSettings
+import com.google.maps.android.compose.rememberCameraPositionState
+import kotlinx.coroutines.launch
+
+/**
+ * Screen for showing a particular route
+ */
+@OptIn(ExperimentalMaterial3Api::class, ExperimentalPermissionsApi::class)
+@Composable
+fun DetailsScreen(navController: NavHostController, homeViewModel: HomeViewModel) {
+
+    val permissionState = rememberPermissionState(Manifest.permission.ACCESS_FINE_LOCATION)
+
+    //Map camera
+    val cameraPositionState = rememberCameraPositionState {
+        position = CameraPosition.fromLatLngZoom(homeViewModel.defaultIthaca, 12f)
+    }
+
+    //Collect flow of route through API
+    val routeApiResponse = homeViewModel.lastRouteFlow.collectAsState().value
+
+    //Map state
+    val mapState = homeViewModel.mapState.collectAsState().value
+
+
+    Column(modifier = Modifier.fillMaxSize()) {
+        //TODO make an AppBarColors class w/ the right colors and correct icon
+        TopAppBar(
+            title = {
+                Text(
+                    text = "Route Details",
+                    fontFamily = sfProDisplayFamily,
+                    fontStyle = FontStyle.Normal
+                )
+            },
+            navigationIcon = {
+                IconButton(onClick = { navController.popBackStack() }) {
+                    Icon(
+                        imageVector = Icons.AutoMirrored.Outlined.KeyboardArrowLeft,
+                        contentDescription = ""
+                    )
+                }
+            }
+
+        )
+
+        Divider(thickness = 1.dp, color = DividerGrey)
+        GoogleMap(
+            modifier = Modifier.fillMaxSize(),
+            cameraPositionState = cameraPositionState,
+            properties = MapProperties(
+                isMyLocationEnabled = permissionState.status.isGranted
+            ),
+            uiSettings = MapUiSettings(zoomControlsEnabled = false)
+        ) {
```

Comment:
To make this composable more readable and a bit less jumbled with everything in one function, let's split the map function and everything in it into its own `private fun`. Take a look at how I (generally) split `HomeScreen.kt` on [this eatery PR.](https://github.com/cuappdev/eatery-blue-android/pull/143)

Feel free to ask for clarification